### ISSUE TITLE
failed to start WinRing0 driver, error 395

### ISSUE THREAD

**Issue by Dark-Hunter540** [xmrig/xmrig#2305]
Here are my logs:
![image](https://user-images.githubusercontent.com/10456133/115936646-ba6d4780-a48d-11eb-9b23-4dca0c0d044d.png)

I cannot manage to get MSR mod to work. Are these related?

**Comment by Spudz76** [comment]
Yes, on Windows it must load the Ring0 driver in order to access CPU MSRs.  Unable to load driver leads directly to not being able to change MSRs.  You must have Secure Boot disabled or MSRs are forever locked.  Could also be something with virus protection not allowing "unknown" vxd's to load.  The VXD file should be in the same folder as the `xmrig.exe` and is called `WinRing0x64.sys`.

Another method is to load Open Hardware Monitor which installs its own MSR driver, but if that works then xmrig will use it (with a complaint there is already some other Ring0 driver, but it uses it fine).  Perhaps OHM installer is smarter about whitelisting its VXD.  Or, if it also doesn't work then it's Secure Boot or some other security junk blocking it.

**Comment by Dark-Hunter540** [comment]
> Yes, on Windows it must load the Ring0 driver in order to access CPU MSRs. Unable to load driver leads directly to not being able to change MSRs. You must have Secure Boot disabled or MSRs are forever locked. Could also be something with virus protection not allowing "unknown" vxd's to load. The VXD file should be in the same folder as the `xmrig.exe` and is called `WinRing0x64.sys`.
> 
> Another method is to load Open Hardware Monitor which installs its own MSR driver, but if that works then xmrig will use it (with a complaint there is already some other Ring0 driver, but it uses it fine). Perhaps OHM installer is smarter about whitelisting its VXD. Or, if it also doesn't work then it's Secure Boot or some other security junk blocking it.

Thank you, it works now. However I am still getting around 4950H/s. Here is what it says now. 
![image](https://user-images.githubusercontent.com/10456133/115961014-6eadb300-a50c-11eb-979b-d042553c3217.png)
Looking at RandomX benchmarks it says I should expect around 12000H/s+. Anything you guys can think of that would be stopping this?

**Comment by clflush** [comment]
what the benchmark didnt capture is the overclocking part. In your case, likely need to overclock and undervolt the CPU, and adjust the DRAM timing.

**Comment by Dark-Hunter540** [comment]
> what the benchmark didnt capture is the overclocking part. In your case, likely need to overclock and undervolt the CPU, and adjust the DRAM timing.

Hi, I have overclocked and undervolted my cpu using PBO2. It regularly boosts over 5GHz. I will overclock my RAM too and get back to you.

**Comment by SChernykh** [comment]
You're still running only 6 threads on 5800X, it should be 16 threads on this CPU. Use xmrig.com/wizard to create new config.

**Comment by Dark-Hunter540** [comment]
> You're still running only 6 threads on 5800X, it should be 16 threads on this CPU. Use xmrig.com/wizard to create new config.

That helped thanks. Creating a new config file has boosted my hashrate to about 8300H/s. I also overclocked my RAM to 3800Mhz. Any ideas on what is preventing it from reaching 12kH/s+? Many thanks so far.
![image](https://user-images.githubusercontent.com/10456133/115963896-85a6d200-a519-11eb-8c97-53491492c890.png)

**Comment by SChernykh** [comment]
> Any ideas on what is preventing it from reaching 12kH/s+?

Maybe other programs running and using CPU. Try setting high performance mode in power options, also go through https://xmrig.com/docs/miner/randomx-optimization-guide/ (Windows 10 tuning guide there).

**Comment by Dark-Hunter540** [comment]
> > Any ideas on what is preventing it from reaching 12kH/s+?
> 
> Maybe other programs running and using CPU. Try setting high performance mode in power options, also go through https://xmrig.com/docs/miner/randomx-optimization-guide/ (Windows 10 tuning guide there).

Hi again,

I carried out the optimisations mentioned. I then proceeded to run the benchmark.
My results are:
https://xmrig.com/benchmark/j2E3s
https://xmrig.com/benchmark/6yGPi
Any tips to improve this?
Thanks

**Comment by RaptorBlue428** [comment]
i have a cannot set msr error and service msr already exists error i have load hardware moniotr

**Comment by RaptorBlue428** [comment]
so help me god
